### PR TITLE
AD5770R: adds new functions, updates device struct, fixes function name type

### PR DESCRIPTION
**Pull request by mike-bradley**
Adds two new driver functions: to get interface status of device and remove device, freeing spi and device memory.
Updates the device struct after writing input, dac values and issuing a SW LDAC command
Fixes typo in function name for set_hw_ladc to ...ldac to match device pin naming.

### CODE REVIEW

**Comment by mirceacaprioru** [comment]
LGTM, could you just fix the error reported by Travis CI?

File: drivers/dac/ad5770r/ad5770r.c [hunk 1]
```diff
@@ -530,24 +545,32 @@ int32_t ad5770r_set_sw_ldac(struct ad5770r_dev *dev,
...
+			AD5770R_SW_LDAC_CH(sw_ldac->en5, AD5770R_CH5);
+
+	ret = ad5770r_spi_reg_write(dev, AD5770R_SW_LDAC, sw_ldac_value);
 
 	if (ret)
 		return ret;
 
 	dev->sw_ldac = *sw_ldac;
 
+        uint8_t  i;
```
Comment (mirceacaprioru) [inline drivers/dac/ad5770r/ad5770r.c]: variable declaration must be at the start of the function, not in the body.

Comment (mirceacaprioru) [reply drivers/dac/ad5770r/ad5770r.c]: move it after 
```
uint8_t sw_ldac_value, i;
```

File: drivers/dac/ad5770r/ad5770r.c [hunk 2]
```diff
@@ -420,16 +420,25 @@ int32_t ad5770r_set_dac_value(struct ad5770r_dev *dev,
...
 	if (!dev)
 		return FAILURE;
 
 	data[1] = (uint8_t)AD5770R_CH_DAC_DATA_LSB(dac_value);
 	data[0] = (uint8_t)((dac_value & 0x3FC0) >> 6);
 
-	return ad5770r_spi_reg_write_multiple(dev,
-					      AD5770R_CH0_DAC_MSB + 2 * channel,
-					      data, sizeof(data));
+	ret = ad5770r_spi_reg_write_multiple(dev,
```
Comment (mirceacaprioru) [inline drivers/dac/ad5770r/ad5770r.c]: i think two commits got squashed here. This commit does more that what it's described in the commit message.

Comment (mike-bradley) [reply drivers/dac/ad5770r/ad5770r.c]: indeed - I'll split back to two separate commits as it was before.

**Pull request merged**